tensorflow-ops/src/TensorFlow/Ops.hs

```diff
@@ -161,6 +164,18 @@ placeholder shape' =
             & opAttr "dtype" .~ tensorType (undefined :: a)
             & opAttr "shape" .~ shape'
 
+
+-- | Construct a tensor whose value is the initialized value of the given
+-- tensor.
+initializedValue :: forall a v.  TensorType a 
+    => Tensor v a 
+    -> Build (Tensor v a)
+initializedValue t = do
+    ns <- use initializationNodes
+    -- Make this tensor depend on the initializers of the other.
+    withNodeDependencies (fromList ns) (render t)
```

Comment:
I'm certainly impressed by how small the change is :)

On the other hand, this is a really big hammer which will impede concurrency (and result in O(n^2) graph size). This effect may also kick in unexpectedly when somebody refactors their code and a variable that was created with few `initializationNodes` suddenly get moved to a later point when a lot of variables are created prior to it.